app/code/Magento/Csp/Test/Unit/Model/Mode/ConfigManagerTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Csp\Test\Unit\Model\Mode;
+
+use Magento\Csp\Model\Mode\ConfigManager;
+use Magento\Csp\Model\Mode\Data\ModeConfigured;
+use Magento\Framework\App\Area;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\State;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Store\Model\Store;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use RuntimeException;
+
+/**
+ * Class ConfigManagerTest
+ *
+ * Test for \Magento\Csp\Model\Mode\ConfigManager
+ */
+class ConfigManagerTest extends TestCase
+{
+    /**
+     * @var ConfigManager
+     */
+    private $model;
+
+    /**
+     * @var ScopeConfigInterface|MockObject
+     */
+    private $scopeConfigMock;
+
+    /**
+     * @var Store|MockObject
+     */
+    private $storeMock;
+
+    /**
+     * @var State|MockObject
+     */
+    private $stateMock;
+
+    /**
+     * Set Up
+     */
+    protected function setUp()
+    {
+        $objectManager = new ObjectManager($this);
+
+        $this->scopeConfigMock = $this->createMock(ScopeConfigInterface::class);
+        $this->storeMock = $this->createMock(Store::class);
+        $this->stateMock = $this->createMock(State::class);
+
+        $this->model = $objectManager->getObject(
+            ConfigManager::class,
+            [
+                'config' => $this->scopeConfigMock,
+                'storeModel' => $this->storeMock,
+                'state' => $this->stateMock
+            ]
+        );
+    }
+
+    /**
+     * Test throwing an exception for non storefront or admin areas
+     *
+     * @return void
+     *
+     * @expectedExceptionMessage CSP can only be configured for storefront or admin area
+     * @expectedException RuntimeException
```

Comment:
Use `$this->expectException` and `$this->expectExceptionMessage` instead.
Annotations are going to be deprecated in further versions of PHPUnit and finally removed in PHPUnit 9.

Reply:
Hi @lbajsarowicz, that's makes sense.
Thanks.